addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/internal/KonnectedHandlerFactory.java

```diff
@@ -0,0 +1,158 @@
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.http.HttpService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link KonnectedHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Zachary Christiansen - Initial contribution
+ */
+@Component(configurationPid = "binding.konnected", service = ThingHandlerFactory.class)
+public class KonnectedHandlerFactory extends BaseThingHandlerFactory {
+    private final Logger logger = LoggerFactory.getLogger(KonnectedHandlerFactory.class);
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_MODULE);
+    private Map<ThingUID, ServiceRegistration<?>> webHookServiceRegs = new HashMap<>();
+    private ServiceRegistration<?> webHookServiceReg;
+    private HttpService httpService;
+    private String callbackUrl = null;
+    private NetworkAddressService networkAddressService;
+    private KonnectedHTTPServlet servlet;
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected void activate(ComponentContext componentContext) {
+        super.activate(componentContext);
+        Dictionary<String, Object> properties = componentContext.getProperties();
+        callbackUrl = (String) properties.get("callbackUrl");
+        KonnectedHTTPServlet servlet = registerWebHookServlet();
+        this.servlet = servlet;
+    }
+
+    @Override
+    protected void deactivate(ComponentContext componentContext) {
+        super.deactivate(componentContext);
+        unregisterWebHookServlet();
+    }
+
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        KonnectedHandler thingHandler = new KonnectedHandler(thing, '/' + BINDING_ID, createCallbackUrl(),
+                createCallbackPort());
+        logger.debug("Adding thinghandler for thing {} to webhook.", thing.getUID().getId());
+        try {
+            servlet.add(thingHandler);
+        } catch (KonnectedWebHookFail e) {
+            logger.trace("there was an error adding the thing handler to the servlet: {}", e.getMessage());
+        }
+        return thingHandler;
+    }
+
+    /**
+     * @param thingHandler thing handler to be removed
+     */
+    @Override
+    protected void removeHandler(ThingHandler thingHandler) {
+        servlet.remove((KonnectedHandler) thingHandler);
+        thingHandler.dispose();
+        super.removeHandler(thingHandler);
+    }
+
+    private KonnectedHTTPServlet registerWebHookServlet() {
+        KonnectedHTTPServlet servlet = null;
+
+        String configCallBack = '/' + BINDING_ID;
+        servlet = new KonnectedHTTPServlet(httpService, configCallBack);
+        if (bundleContext != null) {
+            webHookServiceReg = bundleContext.registerService(HttpServlet.class.getName(), servlet,
```

Comment:
You already register the servlet on the httpService yourself in https://github.com/openhab/openhab2-addons/pull/3702/files#diff-9fd62b4c375582725c471a59f2936986R81, so you must not register it as a servlet here anymore (as this might make white-board-enabled http servers try to register it twice).

Reply:
removed